ext/node/polyfills/worker_threads.ts

```diff
@@ -175,17 +175,17 @@ class NodeWorker extends EventEmitter {
         // empty catch block when package json might not be present
       }
       if (
-        !(StringPrototypeEndsWith(
+        (StringPrototypeEndsWith(
           StringPrototypeToString(specifier),
           ".mjs",
         )) ||
         (pkg && pkg.exists && pkg.typ == "module")
       ) {
+        specifier = toFileUrl(specifier as string);
+      } else {
         const cwdFileUrl = toFileUrl(Deno.cwd());
         specifier =
           `data:text/javascript,(async function() {const { createRequire } = await import("node:module");const require = createRequire("${cwdFileUrl}");require("${specifier}");})();`;
-      } else {
-        specifier = toFileUrl(specifier as string);
       }
```

Comment:
> Nevertheless, a more general solution for the file type detection...

There is a function, `url_to_node_resolution` on `NodeResolver`. Probably that should be used everywhere. Also, all files outside a node_modules directory are considered ESM in Deno (there is a `in_npm_package` function on `NodeResolver` to tell that).

Reply:
The `url_to_node_resolution` handles only an URL to path resp. string conversion.
But it doesn't provide any file type related differentiation, if I'm not wrong.

Right now the format decision is done by looking at 1.) the file suffix and 2.) `op_require_read_closest_package_json` to check, if a `type: module` entry is present somewhere in a `package.json` file above the script.

But **the code without my patch always seemed to choose the exactly opposite consequence based on these checks!** ESM files got wrapped in `require(...)` fragments and CJS files became imported by their URL, not the other way around! (<= But please double-check this claim!)

But `.mjs` files still seemed to work all the time somehow and simply ignored this defect.
That's why the unit test doesn't report an error.

Reply:
> The url_to_node_resolution handles only an URL to path resp. string conversion.
But it doesn't provide any file type related differentiation, if I'm not wrong.

It works for `file:///` urls.

Reply:
That's interesting! 
Which property of the URL object could be used for a more useful import type specific proceeding?

Reply:
@dsherret: I think, I finally understood now your `url_to_node_resolution`! :)

I'll try to use it as replacement here as well to minimize the code duplication and unexpected behavior.